### ISSUE TITLE
Something broken beetween 5.0.1 and ^5.0.0+2 version

### ISSUE THREAD

**Issue by otopba** [OpenFlutter/flutter_screenutil#295]
I didn't change the application code. I just raised the dependency version and my layout changed

version 5.0.0+2:
![telegram-cloud-photo-size-2-5267224222982322032-y](https://user-images.githubusercontent.com/4225279/143689750-122b097a-acf2-48f8-878e-24f9ae972f49.jpg)

version 5.0.1:
![telegram-cloud-photo-size-2-5267224222982322033-y](https://user-images.githubusercontent.com/4225279/143689761-493ad9e8-9952-4052-8e50-0d0b0918559d.jpg)

**Comment by lizhuoyuan** [comment]
请显示代码

**Comment by otopba** [comment]
Oh. This is code of this page

```dart
import 'dart:math';
import 'dart:ui';

import 'package:cached_network_image/cached_network_image.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:word_daily/data/models/common/word.dart';
import 'package:word_daily/domain/share_service.dart';
import 'package:word_daily/logs/log.dart';
import 'package:word_daily/services/preferences/preferences_service.dart';
import 'package:word_daily/ui/bottom_sheet/share_bottom_sheet.dart';
import 'package:word_daily/ui/bottom_sheet/wd_bottom_sheet.dart';
import 'package:word_daily/ui/kit/wd_animations.dart';
import 'package:word_daily/ui/kit/wd_colors.dart';
import 'package:word_daily/ui/kit/wd_images.dart';
import 'package:word_daily/ui/kit/wd_page.dart';
import 'package:word_daily/ui/kit/wd_page_mixin.dart';
import 'package:word_daily/ui/kit/wd_text_style.dart';
import 'package:word_daily/ui/widgets/buttons/bottom_wide_button_with_gradient.dart';
import 'package:word_daily/utils/texts.dart';
import 'package:word_daily/utils/throttler.dart';
import 'package:provider/provider.dart';
import 'package:word_daily/utils/vibrates.dart';

const _tag = 'share_page';

class BackgroundColor {
  final Color color;
  final bool dark;

  BackgroundColor(this.color, this.dark);
}

class SharePage extends StatefulWidget {
  static const routeName = '/share';

  final Word word;

  const SharePage({required this.word, Key? key}) : super(key: key);

  @override
  _SharePageState createState() => _SharePageState();
}

class _SharePageState extends State<SharePage> with WDPageStateMixin {
  static final _backgroundColors = [
    BackgroundColor(const Color(0xFF6955D4), true),
    BackgroundColor(const Color(0xFF000000), true),
    BackgroundColor(const Color(0xFFFFE598), false),
    BackgroundColor(const Color(0xFFFDBBF0), false),
    BackgroundColor(const Color(0xFFFFFFFF), false),
    BackgroundColor(const Color(0xFF91F4B3), false),
    BackgroundColor(const Color(0xFFFE524E), true),
    BackgroundColor(const Color(0xFF45479C), true),
    BackgroundColor(const Color(0xFFACDCFF), false),
  ];
  static const _colorItemSize = 36.0;
  static const _colorItemSpacing = 28.0;
  late final PreferencesService _prefs;
  late final Throttler _closeClickThrottler;
  int _selectedColorIndex = 0;
  final _exportKey = GlobalKey();

  @override
  void initState() {
    super.initState();
    _closeClickThrottler = Throttler(_closePressed);
    _prefs = context.read();
    _selectedColorIndex = max(
      0,
      min(_prefs.getLastShareColorIndex(), _backgroundColors.length - 1),
    );
  }

  @override
  Widget build(BuildContext context) {
    return WDPage(
      padding: EdgeInsets.zero,
      showAppBar: false,
      foreground: _foreground(),
    );
  }

  Widget _foreground() {
    return Column(
      mainAxisSize: MainAxisSize.max,
      children: [
        _topRow(),
        const SizedBox(height: 50),
        _shareContainer(),
        const SizedBox(height: 50),
        _colorList(),
        const SizedBox(height: 14),
        _shareButton(),
      ],
    );
  }

  // Такая матрешка из контейнеров из-за https://github.com/flutter/flutter/issues/37578
  Widget _shareContainer() {
    return Expanded(
      child: Container(
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(25),
          border: Border.all(width: 4, color: colors.listItemBackgroundColor),
        ),
        child: AnimatedContainer(
          duration: WDAnimations.duration,
          curve: WDAnimations.onScreenCurve,
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(21),
            color: _backgroundColors[_selectedColorIndex].color,
          ),
          child: Padding(
            padding: const EdgeInsets.all(6.5),
            child: RepaintBoundary(
              key: _exportKey,
              child: AspectRatio(
                aspectRatio: ShareService.ratio,
                child: ScreenUtilInit(
                  designSize: const Size(
                    ShareService.shareWidth,
                    ShareService.shareHeight,
                  ),
                  builder: () => AnimatedContainer(
                    duration: WDAnimations.duration,
                    curve: WDAnimations.onScreenCurve,
                    color: _backgroundColors[_selectedColorIndex].color,
                    child: _shareContent(),
                  ),
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }

  Widget _shareContent() {
    return Padding(
      padding: EdgeInsets.symmetric(horizontal: 36.w),
      child: Stack(
        children: [
          Column(
            mainAxisSize: MainAxisSize.max,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              SizedBox(height: 200.h),
              _image(),
              SizedBox(height: 93.h),
              _textLine1(),
              if (widget.word.fullTranslatedText.isNullOrEmpty) ...[
                SizedBox(height: 10.h),
                _textLine2(),
              ],
              SizedBox(height: 92.h),
              Opacity(
                opacity: 0.0,
                child: _appDescription(),
              ),
              SizedBox(height: 184.h),
            ],
          ),
          Column(
            mainAxisSize: MainAxisSize.max,
            mainAxisAlignment: MainAxisAlignment.end,
            children: [
              _appDescription(),
              SizedBox(height: 184.h),
            ],
          ),
        ],
      ),
    );
  }

  Widget _textLine1() {
    return SizedBox(
      width: double.maxFinite,
      child: Padding(
        padding: EdgeInsets.symmetric(horizontal: 59.w),
        child: AnimatedDefaultTextStyle(
          duration: WDAnimations.duration,
          curve: WDAnimations.onScreenCurve,
          child: Text(
            (widget.word.fullTranslatedText ?? widget.word.shortText)
                .removeSpaces(),
          ),
          style: TextStyle(
            fontFamily: defaultTargetPlatform == TargetPlatform.iOS ||
                    defaultTargetPlatform == TargetPlatform.macOS
                ? 'SFProDisplay'
                : null,
            fontSize: 68.1.sp,
            letterSpacing: 1.021.sp,
            height: 1.1,
            fontWeight: FontWeight.w600,
            color: _backgroundColors[_selectedColorIndex].dark
                ? WDColors.whiteColor
                : WDColors.blackColor,
          ),
        ),
      ),
    );
  }

  Widget _textLine2() {
    return SizedBox(
      width: double.maxFinite,
      child: Padding(
        padding: EdgeInsets.symmetric(horizontal: 59.w),
        child: AnimatedDefaultTextStyle(
          duration: WDAnimations.duration,
          curve: WDAnimations.onScreenCurve,
          child: Text(
            widget.word.shortTranslatedText.removeSpaces(),
          ),
          style: TextStyle(
            fontFamily: defaultTargetPlatform == TargetPlatform.iOS ||
                    defaultTargetPlatform == TargetPlatform.macOS
                ? 'SFProDisplay'
                : null,
            fontSize: 68.1.sp,
            letterSpacing: 1.021.sp,
            height: 1.1,
            fontWeight: FontWeight.w600,
            color: _backgroundColors[_selectedColorIndex].dark
                ? WDColors.whiteOp50Color
                : WDColors.blackOp40Color,
          ),
        ),
      ),
    );
  }

  Widget _image() {
    return Flexible(
      child: ClipRRect(
        borderRadius: BorderRadius.circular(56.75.r),
        child: CachedNetworkImage(
          errorWidget: (
            BuildContext context,
            String url,
            dynamic error,
          ) {
            Log.e(_tag, "Error loading image $url");
            return Container();
          },
          imageUrl: widget.word.cardImageUrl!,
          fit: BoxFit.fitWidth,
        ),
      ),
    );
  }

  Widget _appDescription() {
    return AnimatedContainer(
      duration: WDAnimations.duration,
      curve: WDAnimations.onScreenCurve,
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(59.r),
        color: _backgroundColors[_selectedColorIndex].dark
            ? WDColors.whiteColor.withOpacity(0.07)
            : WDColors.blackColor.withOpacity(0.04),
      ),
      child: Padding(
        padding: EdgeInsets.only(left: 67.w),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Flexible(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.start,
                mainAxisSize: MainAxisSize.min,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  SizedBox(height: 84.68.h),
                  WDImages.memoTitle(
                    width: 172.8.w,
                    height: 31.68.h,
                    color: _backgroundColors[_selectedColorIndex].dark
                        ? WDColors.whiteColor
                        : WDColors.blackColor,
                  ),
                  SizedBox(height: 23.04.h),
                  FittedBox(
                    fit: BoxFit.scaleDown,
                    child: AnimatedDefaultTextStyle(
                      duration: WDAnimations.duration,
                      curve: WDAnimations.onScreenCurve,
                      child: Text(localizations.appSubtitle),
                      style: TextStyle(
                        fontFamily: defaultTargetPlatform ==
                                    TargetPlatform.iOS ||
                                defaultTargetPlatform == TargetPlatform.macOS
                            ? 'SFProDisplay'
                            : null,
                        fontSize: 43.2128.sp,
                        letterSpacing: 0.915095.sp,
                        height: 1.1,
                        fontWeight: FontWeight.w600,
                        color: _backgroundColors[_selectedColorIndex].dark
                            ? WDColors.whiteColor.withOpacity(0.4)
                            : WDColors.blackColor.withOpacity(0.4),
                        leadingDistribution: TextLeadingDistribution.even,
                      ),
                    ),
                  ),
                ],
              ),
            ),
            Image.asset(WDImages.shareLogo, height: 254.h),
          ],
        ),
      ),
    );
  }

  Widget _topRow() {
    return SafeArea(
      top: true,
      bottom: false,
      left: false,
      right: false,
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        crossAxisAlignment: CrossAxisAlignment.center,
        textBaseline: TextBaseline.alphabetic,
        children: [
          _closeButton(),
          _title(),
          IgnorePointer(
            child: Opacity(
              opacity: 0.0,
              child: _closeButton(),
            ),
          ),
        ],
      ),
    );
  }

  Widget _closeButton() {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 17),
      child: Material(
        type: MaterialType.transparency,
        child: InkWell(
          customBorder: const CircleBorder(),
          radius: 23,
          onTap: _closeClickThrottler.throttle,
          child: Padding(
            padding: const EdgeInsets.all(11),
            child: WDImages.close(
              width: 26,
              height: 26,
              color: colors.primaryTextColor,
            ),
          ),
        ),
      ),
    );
  }

  Widget _title() {
    return Text(
      localizations.share,
      textAlign: TextAlign.center,
      style: WDTextStyle.instance.style18.copyWith(
        color: colors.primaryTextColor,
        leadingDistribution: TextLeadingDistribution.even,
      ),
    );
  }

  Widget _shareButton() {
    return Align(
      alignment: Alignment.bottomCenter,
      child: BottomWideButtonWithGradient(
        gradientColor: colors.backgroundColor,
        centralText: localizations.share,
        centralTextColor: WDColors.blackColor,
        color: WDColors.mainColor,
        onPressed: _onSharePressed,
        maxWidth: MediaQuery.of(context).size.width - 15 * 2,
      ),
    );
  }

  Widget _colorList() {
    final width = MediaQuery.of(context).size.width;
    final capacity =
        (width + _colorItemSpacing) ~/ (_colorItemSize + _colorItemSpacing);
    final padding = max(
        0.0,
        width -
            (capacity * (_colorItemSize + _colorItemSpacing) -
                _colorItemSpacing -
                _colorItemSize / 2.0));

    return SizedBox(
      height: _colorItemSize,
      child: ListView.separated(
        padding: EdgeInsets.only(left: padding, right: 36),
        scrollDirection: Axis.horizontal,
        itemBuilder: _itemBuilder,
        separatorBuilder: _separatorBuilder,
        itemCount: _backgroundColors.length,
      ),
    );
  }

  Widget _itemBuilder(BuildContext context, int index) {
    return Material(
      type: MaterialType.transparency,
      child: InkWell(
        onTap: () => _onColorPressed(index),
        customBorder: const CircleBorder(),
        radius: _colorItemSize / 2,
        child: Container(
          height: _colorItemSize,
          width: _colorItemSize,
          decoration: BoxDecoration(
            color: _backgroundColors[index].color,
            shape: BoxShape.circle,
            border: Border.all(
              width: 1.5,
              color: colors.listItemBackgroundColor,
            ),
          ),
          child: _check(index),
        ),
      ),
    );
  }

  Widget _check(int index) {
    return Center(
      child: AnimatedOpacity(
        opacity: _selectedColorIndex == index ? 1.0 : 0.0,
        duration: WDAnimations.duration,
        child: WDImages.checkSmall(
          width: 26,
          height: 26,
          color: _backgroundColors[_selectedColorIndex].dark
              ? WDColors.whiteColor
              : WDColors.blackColor,
        ),
      ),
    );
  }

  Widget _separatorBuilder(BuildContext context, int index) {
    return const SizedBox(width: _colorItemSpacing);
  }

  void _closePressed() {
    Log.d(_tag, "Close pressed");

    Navigator.of(context, rootNavigator: true).pop();
  }

  void _onSharePressed() {
    Log.d(_tag, "On share pressed");

    WDBottomSheet.showBottomSheet(
      context: context,
      builder: (_) => ShareBottomSheet(
        exportKey: _exportKey,
        word: widget.word,
        backgroundColor: _backgroundColors[_selectedColorIndex].color,
      ),
    );
  }

  void _onColorPressed(int index) {
    Log.d(_tag, "On color pressed $index");

    vibrateSelection();

    setState(() {
      _selectedColorIndex = index;
    });

    _prefs.setLastShareColorIndex(index);
  }
}

```

**Comment by lizhuoyuan** [comment]
Is it a problem with the original size and scaling of the image?

**Comment by otopba** [comment]
Not only that, this block is bigger too. It seems as if the extension (.h) has begun to work differently
![image](https://user-images.githubusercontent.com/4225279/144013985-318602cf-5b12-4514-858b-2662d7e5b665.png)

**Comment by lizhuoyuan** [comment]
I'm not sure if it is caused by the new features of the previous version, you can try the latest version now

**Comment by otopba** [comment]
No, the new version has the same problem =(

**Comment by lizhuoyuan** [comment]
.h always depends on the height of the screen, unless this is changed. Try to replace with .w?

**Comment by AndyHolo** [comment]
Same here,it looks everything use .h become more taller then it should be

**Comment by flutter-f** [comment]
yes. i have also same prob with .h ,
what i will do ?

**Comment by Saidjamol** [comment]
Set splitScreenMode: false

**Comment by lizhuoyuan** [comment]
change to new version

**Comment by flutter-f** [comment]
now im using 5.0.0+3 version and i set splitScreenMode: false.
i tried this screen on my device (1440 x 2560) and it worked correctly, but when i tried on device (480 x 800) error happened.
i set these codes:
Image.asset(image, height: 332.h) // for image
Text(title, size: 40.sp) // for title
Text(description, size: 16.sp) // for description.

so i think we have issue in (.sp) , can you please help me ?
thank you very much.

**Comment by flutter-f** [comment]
now im using 5.0.0+3 version and i set splitScreenMode: false.
i tried this screen on my device (1440 x 2560) and it worked correctly, but when i tried on device (480 x 800) error happened.
i set these codes:
Image.asset(image, height: 332.h) // for image
Text(title, size: 40.sp) // for title
Text(description, size: 16.sp) // for description.

so i think we have issue in (.sp) , can you please help me ?
thank you very much.
![kkk](https://user-images.githubusercontent.com/58072084/147354502-c74a3b79-66f6-4aef-ac25-198f2a46108f.PNG)
![kk](https://user-images.githubusercontent.com/58072084/147354513-f75f9a1a-29a1-4fde-b660-d89770e35394.PNG)

**Comment by otopba** [comment]
@lizhuoyuan Tested it on version 5.0.3. Everything works

**Comment by otopba** [comment]
In version 5.1.1 this error appeared again

**Comment by otopba** [comment]
Fixed at 5.2.0
